src/selectors/entities/channels.ts

```diff
@@ -180,15 +182,26 @@ export function filterChannels(
 // - The status of the other user in a DM channel
 export function makeGetChannel(): (state: GlobalState, props: {id: string}) => Channel {
     return createSelector(
-        getAllChannels,
-        (state: GlobalState, props: {id: string}) => props.id,
-        (state: GlobalState) => state.entities.users,
-        getTeammateNameDisplaySetting,
-        (allChannels, channelId, users, teammateNameDisplay) => {
-            const channel = allChannels[channelId];
+        getCurrentUserId,
+        (state: GlobalState) => state.entities.users.profiles,
+        (state: GlobalState) => state.entities.users.profilesInChannel,
+        (state: GlobalState, props: {id: string}) => {
+            const channel = getChannel(state, props.id);
+            if (!channel || !isDirectChannel(channel)) {
+                return '';
+            }
 
+            const currentUserId = getCurrentUserId(state);
+            const teammateId = getUserIdFromChannelName(currentUserId, channel.name);
+            const teammateStatus = getStatusForUserId(state, teammateId);
+
+            return teammateStatus || 'offline';
+        },
+        (state: GlobalState, props: {id: string}) => getChannel(state, props.id),
+        getTeammateNameDisplaySetting,
+        (currentUserId, profiles, profilesInChannel, teammateStatus, channel, teammateNameDisplay) => {
```

Comment:
As mentioned in my other comment, this will still recalculate when any user changes, but this presumably should be better than recalculating whenever anything in `state.entities.users` like statuses changes

Reply:
Yeah, in my testing it was specifically the status changes (which are one of the reducers that change wholesale vs. discretely). in a system with lots of profile changes, this would continue to be a problem but I'm hoping that's less frequent.